src/Qir/Runtime/public/QubitManager.hpp

```diff
@@ -40,7 +40,7 @@ namespace Quantum
         // No complex scenarios for now. Don't need to support copying/moving.
         CQubitManager(const CQubitManager&) = delete;
         CQubitManager& operator=(const CQubitManager&) = delete;
-        virtual ~CQubitManager();
+        ~CQubitManager();
```

Comment:
This is very suspicious and against recommendations. Can you write your experience with crashes in comments or file it as an issue.

Reply:
As for recommendations. Yes, I know there is a recommendation that encourages making the destructors _virtual_. But there is also a refinement (to that recommendation) that instructs to avoid making the destructor virtual if there is no any other virtual function, see [Item 7: Declare destructors virtual in polymorphic base classes](https://aovst.cs.msu.ru/morda/effective_cpp.pdf) (search for "If a class does not contain virtual functions" on paper page 41).
On the other hand, yes, I agree that there is nothing wrong with the code. There is nothing resulting in a crash if the destructor is the only virtual member function of the class.
I'll add the comment in the code.

Reply:
```c++
        ~CQubitManager();   // If this dtor is made _virtual_ then the QIR RT tests crash (at least in Debug config)
          // if the native simulator is compiled with Clang (as opposed to GCC). Nothing wrong found in the code,
          // probably is the compiler bug.
```